libraries/AP_DAL/AP_DAL_Baro.cpp

```diff
@@ -0,0 +1,36 @@
+#include "AP_DAL_Baro.h"
+
+#include <AP_Logger/AP_Logger.h>
+#include "AP_DAL.h"
+
+AP_DAL_Baro::AP_DAL_Baro()
+{
+    for (uint8_t i=0; i<BARO_MAX_INSTANCES; i++) {
+        _RBRI[i].instance = i;
+    }
+}
+
+void AP_DAL_Baro::start_frame()
+{
+    const auto &baro = AP::baro();
+
+    const log_RBRH old_RBRH = _RBRH;
+    _RBRH.primary = baro.get_primary();
+    _RBRH.num_instances = baro.num_instances();
+    WRITE_REPLAY_BLOCK_IFCHANGD(RBRH, _RBRH, old_RBRH);
+
+    for (uint8_t i=0; i<BARO_MAX_INSTANCES; i++) {
```

Comment:
I wonder if this really should be `baro.num_instances()` rather than `BARO_MAX_INSTANCES`.

Similarly elsewhere.

Reply:
done for baro and compass